### PR TITLE
Include gulpfile and build integration config

### PR DESCRIPTION
**Pull request by oreoluwade**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,64 @@
...
+- Jasmine (Test Runner)
+- Travis CI (For Continous Integration and badge)
+- Coveralls (To compute Test Coverage percentage and add a badge that displays the coverage percentage)
+- Hound (To prevent style violations)
+- AngularJs (For making features of the view responsive)
+- Bootstrap (For Styling the view)
+
+
+## Application Limitations
+
```
Comment (oredavids) [inline README.md]: This header has no information provided beneath.

File: jasmine/spec/inverted-index-test.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+// Inverted Index test suite
+describe('Inverted Index Suite', () => {
+  // Define variables to be used in the test Suite
+  const newIndex = new InvertedIndex();
+  const emptyBook = [];
+  const sampleSentence =
+    'This be an %example of the #typical be sentence type.';
+  const multipleSearch = 'Destroy world imagination quickly.';
+  newIndex.createIndex(books);
```
Comment (oredavids) [inline jasmine/spec/inverted-index-test.js]: I believe only line 16 index gets stored with your present index creation structure(line 15 will be overwritten).

File: package.json [hunk 3]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "inverted-index-master",
   "version": "1.0.0",
-  "description": "An app that returns the location of words from a JSON file containing JSON objects ",
-  "main": "gulpfile.js",
+  "description": "An app that returns the location of words from a JSON file containing JSON objects",
+  "main": "app.js",
```
Comment (oredavids) [inline package.json]: Your main file should be the file you use to start your application. In this case; your gulpfile not app.js.

File: package.json [hunk 4]
```diff
@@ -25,19 +26,52 @@
...
+    "coveralls": "^2.11.15",
+    "eslint": "^3.14.0",
     "eslint-config-airbnb": "^14.0.0",
+    "eslint-config-standard": "^6.2.1",
     "eslint-plugin-import": "^2.2.0",
-    "eslint-plugin-jsx-a11y": "^3.0.2"
+    "eslint-plugin-jsx-a11y": "^3.0.2",
+    "eslint-plugin-promise": "^3.4.0",
+    "eslint-plugin-react": "^6.9.0",
+    "eslint-plugin-standard": "^2.0.1",
```
Comment (oredavids) [inline package.json]: I believe your eslint node packages should be in your dev-dependencies, based on your configured tasks, as you would not require them to start your application.

File: src/inverted-index.js [hunk 5]
```diff
@@ -52,7 +54,7 @@ class InvertedIndex {
     const distinctContent = InvertedIndex.distinctWords(wordsToIndex.join(' '));
     distinctContent.forEach((word) => {
       index[word] = [];
-      wordsToIndex.forEach((document, indexPosition) => {
+      [...wordsToIndex].forEach((document, indexPosition) => {
```
Comment (oredavids) [inline src/inverted-index.js]: I believe you can use a forEach on this array without the use of spread.

File: src/app.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+          setMessage('This is not a JSON file.');
+          return;
+        }
+        try {
+          const acceptedJsonFile = JSON.parse(err.target.result);
+          if (acceptedJsonFile.length === 0 || !acceptedJsonFile[0].title ||
+            !acceptedJsonFile[0].text) {
+            $scope.uploadSuccess = false;
+            setMessage('This is an Empty JSON File');
+            $scope.$apply();
```
Comment (emmyakin) [inline src/app.js:37]: setMessage calls $scope.$apply, this may be redundant

**Pull request merged**